web/components/pageComponents/DatabasePage/component.tsx

```diff
@@ -21,6 +31,52 @@ type Props = {
 };
 
 export default function DatabasePage({ params, children, ...props }: Props) {
+  const router = useRouter();
+  const { defaultBranchName } = useDefaultBranch(params);
+  // route to the correct page based on the menu item clicked
+  if (process.env.NEXT_PUBLIC_FOR_ELECTRON === "true") {
+    window.ipc.onMenuClicked(async (value: string) => {
+      const paramsWithRef = {
+        ...params,
+        refName: params.refName || defaultBranchName,
+      };
+      switch (value) {
+        case "upload-file": {
+          const { href, as } = upload(paramsWithRef);
+          router.push(href, as).catch(console.error);
+          break;
+        }
+        case "commit-graph": {
+          const { href, as } = commitGraph(paramsWithRef);
+          router.push(href, as).catch(console.error);
+          break;
+        }
+        case "schema-diagram": {
+          const { href, as } = schemaDiagram(paramsWithRef);
+          router.push(href, as).catch(console.error);
+          break;
+        }
+        case "new-table": {
+          const { href, as } = createTable(paramsWithRef);
+          router.push(href, as).catch(console.error);
+          break;
+        }
+        case "new-branch": {
+          const { href, as } = newBranch(paramsWithRef);
+          router.push(href, as).catch(console.error);
+          break;
+        }
+        case "new-release": {
+          const { href, as } = newRelease(paramsWithRef);
+          router.push(href, as).catch(console.error);
+          break;
+        }
+        default:
+          break;
+      }
+    });
+  }
```

Comment:
This seems like it should be extracted into a hook or util. `useElectronMenu` or something